pkg/machine/connection/connection.go

```diff
@@ -85,37 +83,51 @@ func UpdateConnectionIfDefault(rootful bool, name, rootfulName string) error {
 	})
 }
 
-func RemoveConnections(names ...string) error {
-	return config.EditConnectionConfig(func(cfg *config.ConnectionsFile) error {
-		for _, name := range names {
-			if _, ok := cfg.Connection.Connections[name]; ok {
-				delete(cfg.Connection.Connections, name)
-			} else {
-				return fmt.Errorf("unable to find connection named %q", name)
-			}
+func RemoveConnections(machines map[string]bool, names ...string) error {
+	var dest config.Destination
+	var service string
 
-			if cfg.Connection.Default == name {
-				cfg.Connection.Default = ""
-			}
-		}
-		for service := range cfg.Connection.Connections {
-			cfg.Connection.Default = service
-			break
-		}
-		return nil
-	})
+	if err := config.EditConnectionConfig(func(cfg *config.ConnectionsFile) error {
+		return setNewDefaultConnection(cfg, &dest, &service, names...)
+	}); err != nil {
+		return err
+	}
+
+	rootful, ok := machines[service]
+	if dest.IsMachine && ok {
+		return UpdateConnectionIfDefault(rootful, service, service+"-root")
+	}
+
+	return nil
 }
 
-// removeFilesAndConnections removes any files and connections with the given names
-func RemoveFilesAndConnections(files []string, names ...string) {
-	for _, f := range files {
-		if err := os.Remove(f); err != nil && !errors.Is(err, os.ErrNotExist) {
-			logrus.Error(err)
+// setNewDefaultConnection iterates through the available system connections and
+// sets the first available connection as the new default
+func setNewDefaultConnection(cfg *config.ConnectionsFile, dest *config.Destination, service *string, names ...string) error {
```

Comment:
does anyone think it is worth checking if there is a podman-machine-default and setting that as the default; else the first one we find ?

Reply:
I think that could be worth while. It could be a valid assumption that if the user's current default connection is associated with a machine, they would expect the new default to be as well (if possible).